### PR TITLE
Add cutoff for dry/wet creep in water fugacity calculation

### PR DESCRIPTION
**Pull request by danieldouglas92**
Note that this PR is an extension of PR #5805, it takes the test implemented in that PR and modifies it slightly to serve as a test for the functionality implemented in this PR.

If there is 0 bound water in the model domain and the user tries using the HK04 olivine hydration viscosity prefactor scheme there is a divide by 0 error. In Hirth & Kohlstaedt 2004, they discuss that when the hydration state of olivine is below 50 H/Si ppm (6.15e-4 wt% water), olivine deformation is consistent with dry conditions, and above 6.15e-4 wt% water olivine deformation is consistent with wet conditions. This PR adds a user parameter that defines this cutoff between dry and wet deformation, with the default value representing the value discussed in Hirth & Kohlstaedt 2004.

### CODE REVIEW

File: source/material_model/rheology/compositional_viscosity_prefactors.cc [hunk 1]
```diff
@@ -84,6 +84,14 @@ namespace aspect
       void
       CompositionalViscosityPrefactors<dim>::declare_parameters (ParameterHandler &prm)
       {
+        prm.declare_entry ("Minimum weight percent bound water content for fugacity", "6.15e-6",
+                           Patterns::List(Patterns::Double(0)),
+                           "The minimum water content for the HK04 olivine hydration Viscosity "
```
Comment (naliboff) [inline source/material_model/rheology/compositional_viscosity_prefactors.cc:89]:
```suggestion
                           "The minimum water content for the HK04 olivine hydration viscosity "
```

**Review by naliboff — COMMENTED** [review]
@danieldouglas92 - thanks for the fix. Only one small suggested change (capitalization change). If it would help to get this merged for other PRs you are working on today, I'm happy to approve and then just submit a follow-up PR fixing the capitalization (everything else looks correct).

**Review by naliboff — APPROVED** [review]

**Comment by naliboff** [comment]
@bobmyhill - this looks good from my end if you would like to take a final look before merging.

**Review by bobmyhill — CHANGES_REQUESTED** [review]
@danieldouglas92 the code is fine, but I think the user variable name is incorrect. 

Also, the source code would be easier to read if the variable names followed the user variable names more closely. I've given some suggestions.

**Review by bobmyhill — APPROVED** [review]

**Pull request merged**